pennylane_qiskit/converter.py

```diff
@@ -192,30 +193,28 @@ def _function(params: dict = None, wires: list = None):
                 # Qiskit ParameterExpression, then replace it with the corresponding PennyLane
                 # variable from the var_ref_map dictionary.
 
-                parameters = []
-                for p in op[0].params:
+                pl_parameters = []
+                for p in op.params:
 
                     _check_parameter_bound(p, var_ref_map)
 
                     if isinstance(p, ParameterExpression):
-                        if p.parameters:
-                            # p.parameters must be a single parameter, as PennyLane
-                            # does not support expressions of variables currently.
-                            if len(p.parameters) > 1:
-                                raise ValueError(
-                                    "Operation {} has invalid parameter {}. PennyLane does not support "
-                                    "expressions containing differentiable parameters as operation "
-                                    "arguments".format(instruction_name, p)
-                                )
-
-                            param = min(p.parameters)
-                            parameters.append(var_ref_map.get(param))
+                        if p.parameters:  # non-empty set = has unbound parameters
+                            ordered_params = tuple(p.parameters)
+
+                            f = lambdify(ordered_params, p._symbol_expr, modules=qml.numpy)
```

Comment:
I wonder if this has any performance implications. :thinking: In Strawberry Fields, there was a case where using `lambdify` turned out to not be the best choice:
https://github.com/XanaduAI/strawberryfields/pull/579

Reply:
Acoording to [the sympy docs](https://docs.sympy.org/latest/modules/numeric-computation.html), `lambdify` is more efficient than `evalf`.  Also, it allows us to use `qml.numpy` as the namespace for conversion.  Using our wrapped numpy allows the expression to be differentiable.

I don't think this conversion is ever going to be very much of a bother compared to other bottlenecks.  Qiskit does not seem very supportive of users creating elaborate expressions.

Reply:
Could we keep the condition as before and use the solution with lambdify only `if len(p.parameters) > 1`? That way we'll make use of it fewer times.

Reply:
What is the expression is something like `ParameterExpression(2*a)`.  Then there's only one parameter, `a`, but we can't just use `a`'s value in place of the expression.  This is a case that would have fallen through the cracks before.  If there's a viable alternative that's faster, I'm all for it, but I think this is the only way to support necessary functionality.